app/src/main/java/at_challenge/deaguiar/daniel/atchalleng/util/HttpRequesAsyncTask.java

```diff
@@ -0,0 +1,84 @@
+package at_challenge.deaguiar.daniel.atchalleng.util;
+
+import android.os.AsyncTask;
+import android.util.Base64;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.DefaultHttpClient;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+/**
+ * HttpRequesAsyncTask
+ * Request data from server
+ *
+ * @author Daniel Besen de Aguiar
+ */
+public abstract class HttpRequesAsyncTask extends AsyncTask<Void, Void, String> {
+    protected final static String USERNAME = "WKD4N7YMA1uiM8V";
+    protected final static String PASSWORD = "DtdTtzMLQlA0hk2C1Yi5pLyVIlAQ68";
+
+    protected String mJsonRoute;
+    protected String mEndPoint;
+
+    @Override
+    protected String doInBackground(Void... params) {
+        try {
+            HttpResponse response = getHttpResponse(mJsonRoute);
+
+            String result = "";
+            InputStream inputStream = response.getEntity().getContent();
+            if (inputStream != null) {
+                result = convertInputStreamToString(inputStream);
+            }
+
+            return result;
+        } catch (IOException e) {
+            return "{\"error\": {\"message\": \"Could not connect to Server. Please check your network connection\"}}";
+        }
+    }
+
+    /**
+     * Tries to fetch JSON values from url
+     * @param jsonValue server request
+     * @return server response
+     * @throws IOException
+     */
+    protected HttpResponse getHttpResponse(String jsonValue) throws IOException {
```

Comment:
Would create a class named HttpRequest and put the methods and would put
the method getHttpResponse(...) inside it. Maybe create a class to handle
the conversion of inputStream to String would be good as well.

2015-06-17 13:21 GMT-03:00 André Luiz Cardoso <notifications@github.com>:

> In
> app/src/main/java/at_challenge/deaguiar/daniel/atchalleng/util/HttpRequesAsyncTask.java
> <https://github.com/daniboy000/at_challenge/pull/1#discussion_r32644977>:
>
> > +                result = convertInputStreamToString(inputStream);
> > +            }
> > +
> > +            return result;
> > +        } catch (IOException e) {
> > +            return "{\"error\": {\"message\": \"Could not connect to Server. Please check your network connection\"}}";
> > +        }
> > +    }
> > +
> > +    /**
> > +     * Tries to fetch JSON values from url
> > +     * @param jsonValue server request
> > +     * @return server response
> > +     * @throws IOException
> > +     */
> > +    protected HttpResponse getHttpResponse(String jsonValue) throws IOException {
>
> This code is duplicated. How would you fix that?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/daniboy000/at_challenge/pull/1/files#r32644977>.
>



-- 
Daniel Besen de Aguiar

"Those who wish to succeed must ask the right preliminary questions"


      Aristotle